util/src/main/java/org/datacommons/util/StatChecker.java

```diff
@@ -256,21 +271,46 @@ private int getSamplePlaceBucketSize(String placeDcid) {
     }
   }
 
+  protected static void checkSeriesTypeInconsistencies(
+      List<DataPoint> timeSeries, StatValidationResult.Builder resBuilder, LogWrapper logCtx) {
+    StatValidationEntry.Builder inconsistentTypeCounter = StatValidationEntry.newBuilder();
+    String counterKey = "StatsCheck_Inconsistent_Types";
+    inconsistentTypeCounter.setCounterKey(counterKey);
+
+    ValueType firstType = null;
+
+    for (DataPoint dp : timeSeries) {
+      for (DataValue val : dp.getValuesList()) {
+        ValueType type = val.getValue().getType();
+        if (firstType == null) {
+          firstType = type;
+        }
+        if (!firstType.equals(type)) {
+          inconsistentTypeCounter.addProblemPoints(dp);
+          logCtx.incrementWarningCounterBy(counterKey, 1);
+        }
```

Comment:
We probably want a pair of inconsistent points.  How about we add the first point when "firstType == null" and when you find "firstType != type", also add that point, add to resBuilder and return....

Reply:
Done.